Content.Client/UserInterface/Systems/Actions/Controls/ActionButton.cs

```diff
@@ -195,10 +201,12 @@ private void OnUnpressed(GUIBoundKeyEventArgs args)
         var name = FormattedMessage.FromMarkupPermissive(Loc.GetString(metadata.EntityName));
         var decr = FormattedMessage.FromMarkupPermissive(Loc.GetString(metadata.EntityDescription));
 
-        if (_action is { Charges: not null })
+        if (_entities.TryGetComponent(ActionId, out Shared.Charges.Components.LimitedChargesComponent? actionCharges))
         {
-            var charges = FormattedMessage.FromMarkupPermissive(Loc.GetString($"Charges: {_action.Charges.Value.ToString()}/{_action.MaxCharges.ToString()}"));
-            return new ActionAlertTooltip(name, decr, charges: charges);
+            var charges = _sharedChargesSys.GetCurrentCharges((ActionId.Value, actionCharges, null));
+
+            var chargesText = FormattedMessage.FromMarkupPermissive(Loc.GetString($"Charges: {charges.ToString()}/{actionCharges.MaxCharges}"));
+            return new ActionAlertTooltip(name, decr, charges: chargesText);
```

Comment:
```suggestion
            var charges = _sharedChargesSys.GetCurrentCharges((ActionId.Value, actionCharges, null));
            var chargesText = FormattedMessage.FromMarkupPermissive(Loc.GetString($"Charges: {charges.ToString()}/{actionCharges.MaxCharges}"));

            if (_entities.TryGetComponent(ActionId, out AutoRechargeComponent? autoRecharge))
            {
                var chargeTimeRemaining = _sharedChargesSys.GetNextRechargeTime((ActionId.Value, actionCharges, autoRecharge));
                chargesText.AddText(Loc.GetString($"{Environment.NewLine}Time Til Recharge: {chargeTimeRemaining}"));
            }

            return new ActionAlertTooltip(name, decr, charges: chargesText);
```

This way we relay the info back to the user how long the next charge will be.

![image](https://github.com/user-attachments/assets/3c07286e-7cd7-49c0-a409-c08b6422eb4b)

![image](https://github.com/user-attachments/assets/3aa09cc5-1f4b-4c6d-8d88-f3a5e4c37112)

Reply:
This shouldn't even be hardcoded in here to begin with?

Reply:
Can you explain more why you think this is hardcoded here?
This wouldn't be changed in the action refactor I have planned unless I had someone to work on an actual animation for charges & recharge time on the refactor.

What really is the issue of showing people the information they need instead of keeping it hidden? I do not want recharge time to _not_ be conveyed to the user.

Reply:
For future archaeologists tl;dr is that any number of systems may want to modify tooltips like itemstatus, alerts, or actual tooltips, but for now it's just being slapped in.